qiskit/pulse/instructions/delay.py

```diff
@@ -50,12 +50,14 @@ def __init__(self, duration: int,
             channel: The channel that will have the delay.
             name: Name of the delay for display purposes.
         """
-        if channel is None:
+        self._channel = channel
+        if not isinstance(channel, Channel):
```

Comment:
What's the idea, here? Should not passing a channel be treated the same way as passing a bad argument?
```
sched += Delay(10, "drive0")
```

Reply:
This is for the deprecated case. There was a test like this that wasn't throwing a warning.